### PR TITLE
Adds cob spec code to project

### PR DESCRIPTION
**Pull request by RabeaGleissner**
@felipesere @christophgockel @gemcfadyen

I added the cob spec code to my project and set the routes relative to the location of my jar. So problems with trailing slashes can't happen anymore now! Readme is updated accordingly.

I also started testing an exception but I'm not sure if this is the correct way. Georgina agreed to have a look with me tomorrow morning, so I will continue working on dealing with the exceptions after that.

### CODE REVIEW

**Comment by RabeaGleissner** [comment]
@jsuchy @felipesere @christophgockel @gemcfadyen

I added tests for some of the Socket exceptions in the `Network` class to this PR. 

I'd like to focus on continuing to implement the remaining server responses as a priority for now, hence I haven't tested all of the exceptions in my code.

File: src/test/java/de/rabea/server/NetworkStub.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    public String read() {
+       return message;
+    }
+
+    public void write(String response, byte[] body) {
+        returnedResponse = response + asString(body);
+    }
+
+    public void close() throws IOException {
+        if (exception.equals("throw exception")) {
```
Comment (christophgockel) [inline src/test/java/de/rabea/server/NetworkStub.java]: A string comparison seems rather unusual here.
Why not have something more explicit like a named constructor? So you could call `new NetworkStub.thatThrows()` in a test.
The problem with the string is, _I_ as a user of this double now need to know exactly one particular string value to use it in a correct way.

File: src/test/java/de/rabea/server/NetworkTest.java [hunk 2]
```diff
@@ -39,32 +43,86 @@ public void readsPutRequestWithoutBody() {
...
+        Network network = new Network(socket);
+        network.createSender();
+    }
+
+    public class SocketWithException extends Socket {
+
+        private String method;
+
+        public SocketWithException(String method) {
+            this.method = method;
```
Comment (christophgockel) [inline src/test/java/de/rabea/server/NetworkTest.java]: See above. String configuration for doubles.

**Comment by christophgockel** [comment]
I have to stop commenting here because I can not distinguish between actual refactorings and the noise the vendored CobSpec introduces here (in the diff).
Personally I would have added a CobSpec into a vendor directory outside of any PR or in a separate one.

File: src/test/java/de/rabea/server/NetworkStub.java (hunk 1 above)
Comment (RabeaGleissner) [inline src/test/java/de/rabea/server/NetworkStub.java]: Thanks @christophgockel for your feedback. 
I really like this! I've refactored my code to use this constructor method.

**Comment by RabeaGleissner** [comment]
@christophgockel Sorry about the confusing PR. Yes, you're right, I should have separated adding the framework from my actual code changes.

**Pull request merged**